### PR TITLE
Refactored to components

### PR DESCRIPTION
**Pull request by solomojb**
Hello Again,

  This is a pretty big change. I apologize, i might have gone a little overboard with it, but i think you'll see it'll make adding things a lot simpler going forward.

  At your convenience please review and test. 

Jim

### CODE REVIEW

**Comment by heisch** [comment]
I didn't forget this I'm just super busy at the moment. Will have a look at this soon™.

**Comment by solomojb** [comment]
No worries. I promise future changes will be smaller feature sized improvements.

Working on “party mode” now. Almost have it working.

Also created the ability to filter multiple slots and added for the most part the fallen circle cards. (I didn’t fully track down their sources hoping the community will help)

Sent from my iPhone

On Jul 16, 2020, at 5:17 AM, Nicola Heisch <notifications@github.com> wrote:

﻿

I didn't forget this I'm just super busy at the moment. Will have a look at this soon™.

File: .gitmodules [hunk 1]
```diff
@@ -1,3 +1,3 @@
 [submodule "vendor/any2cards"]
 	path = vendor/any2cards
-	url = git@github.com:any2cards/gloomhaven.git
+	url = https://github.com/any2cards/gloomhaven.git
```
Comment (heisch) [inline .gitmodules:3]: any particular reason for this change?

File: src/components/Tabs/SpoilerFilters/SpoilerFilterItemList.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    end: number;
+    title: string;
+}
+
+const SpoilerFilterItemList = (props:Props) => {
+    const {start, end, title} = props;
+
+    const checkBoxes = [];
+    for (let i = start; i <= end; i++) {
+        checkBoxes.push(<FilterCheckbox id={i}/>);
```
Comment (heisch) [inline src/components/Tabs/SpoilerFilters/SpoilerFilterItemList.tsx]: Each child in a list should have a unique "key" prop.

File: src/components/Tabs/MainView/ItemCard.tsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+            <img
+                src={getItemImageSrc()}
+                alt={item.name}
+                className={'item-card'}/>
+            <ItemManagement item={item}/>
+        </div>
+    )
+}
+
+export default ItemCard;
```
Comment (heisch) [inline src/components/Tabs/MainView/ItemCard.tsx]: Can you set up your editor/IDE to always add a newline at the end of a file when saving?

File: src/components/Tabs/MainView/ItemGrid.tsx [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+import ItemCard from './ItemCard';
+
+type Props = {
+    items : GloomhavenItem[];
+}
+
+const ItemGrid = (props:Props) => {
+    const {items} = props;
+    return (
+        <>
```
Comment (heisch) [inline src/components/Tabs/MainView/ItemGrid.tsx:12]: Didn't know there is a short syntax for <React.Fragment> cool

File: src/components/Tabs/MainView/ItemList.tsx [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+        const setSorting = (newProperty: SortProperty) => {
+            let newDirection:SortDirection;
+            if (property === newProperty) {
+                newDirection = direction === SortDirection.ascending ? SortDirection.descending : SortDirection.ascending;
+            } else {
+                newDirection = SortDirection.ascending;
+            }
+
+            console.log(newProperty, newDirection);
```
Comment (heisch) [inline src/components/Tabs/MainView/ItemList.tsx]: forgotten debug statement

File: src/components/Tabs/MainView/ItemManagement.tsx [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+const ItemManagement = (props:Props) => {
+    const {item} = props;
+    const { enableStoreStockManagement, lockSpoilerPanel, itemsInUse } = useSelector<RootState>( state => state.spoilerFilter) as RootState['spoilerFilter'];
+    const dispatch = useDispatch();
+
+    if (!enableStoreStockManagement) {
+        return <>
+                {item.count};
+                </>
```
Comment (heisch) [inline src/components/Tabs/MainView/ItemManagement.tsx]: Please always use brackets for multi-line html
```js
return (
  <>
    {item.count};
  </>
);
```

**Review by heisch — CHANGES_REQUESTED** [review]
Other than some minor nitpicks this looks very good.
Please address the issues and then you can tell me when this is ready to publish

File: .gitmodules (hunk 1 above)
Comment (solomojb) [reply .gitmodules:3]: I updated to the head of the submodule to get access to the Fallen Circle Cards. (Which will be in a future branch)

File: src/components/Tabs/SpoilerFilters/SpoilerFilterItemList.tsx (hunk 2 above)
Comment (solomojb) [reply src/components/Tabs/SpoilerFilters/SpoilerFilterItemList.tsx]: Added

File: src/components/Tabs/MainView/ItemCard.tsx (hunk 3 above)
Comment (solomojb) [reply src/components/Tabs/MainView/ItemCard.tsx]: Done!

File: src/components/Tabs/MainView/ItemList.tsx (hunk 5 above)
Comment (solomojb) [reply src/components/Tabs/MainView/ItemList.tsx]: oops good catch

**Comment by solomojb** [comment]
I pushed the fixes to the code.

**Review by heisch — APPROVED** [review]

**Pull request merged**